tests/compile-fail/ptr_int_guess.rs

```diff
@@ -0,0 +1,11 @@
+// compile-flags: -Zmiri-permissive-provenance -Zmiri-disable-stacked-borrows
+
+fn main() {
+    let x: i32 = 3;
+    let x_ptr = &x as *const i32;
+
+    // TODO: switch this to addr() once we intrinsify it
+    let x_usize: usize = unsafe { std::mem::transmute(x_ptr) };
+    let ptr = x_usize as *const i32;
```

Comment:
```suggestion
    // Cast back a pointer that did *not* get exposed.
    let ptr = x_usize as *const i32;
```
Maybe also put "not_exposed" into the filename, that seems more clear than "guess".

(I think we need a "pointers" folder in compile-fail... but that is for another PR.)